### ISSUE TITLE
Crash while loading a page with a lot of resources.

### ISSUE THREAD

**Issue by Twibit** [readium/SDKLauncher-Android#23]
Hello,

I have an epub with a lot of image and audio resources.
When loading a page,  the app crashes with the following error while loading a resource with createResourceInputStream. 

 ```
1662   libepub3 [./../../..  D  StaticMethod(): calling static method 'createResourceInputStream[(JJ)Lorg/readium/sdk/android/util/ResourceInputStream;]'
  121                   DEBUG  I  *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
  121                   DEBUG  I  Build fingerprint: 'google/nakasi/grouper:4.4.2/KOT49H/937116:user/release-keys'
  121                   DEBUG  I  Revision: '0'
  121                   DEBUG  I  pid: 1662, tid: 1793, name: NanoHttpd Reque  >>> org.readium.sdk.android.launcher <<<
  121                   DEBUG  I  signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0000001c
  121                   DEBUG  I  r0 69ed6100  r1 0018619e  r2 00000000  r3 00000000
  121                   DEBUG  I  r4 69ed6100  r5 6a644ef8  r6 00000000  r7 6e523e1c
  121                   DEBUG  I  r8 6f39bb0c  r9 6e523e10  sl 6a644f08  fp 6f39bb24
  121                   DEBUG  I  ip 00000006  sp 6f39baf8  lr 6b227b23  pc 6b227b26  cpsr 60030030
  121                   DEBUG  I  d0  0000000000000000  d1  0000000000000000
  121                   DEBUG  I  d2  6165727453747570  d3  6f4c294a4a285b6d
  121                   DEBUG  I  d4  69646165722f6772  d5  612f6b64732f6d75
  121                   DEBUG  I  d6  752f64696f72646e  d7  6f7365522f6c6974
  121                   DEBUG  I  d8  4388000000000110  d9  000000004387e600
  121                   DEBUG  I  d10 0000000000000000  d11 0000000000000000
  121                   DEBUG  I  d12 0000000000000000  d13 0000000000000000
  121                   DEBUG  I  d14 0000000000000000  d15 0000000000000000
  121                   DEBUG  I  d16 000000000000001c  d17 0000000000000000
  121                   DEBUG  I  d18 3fd0000000000000  d19 3ffe6c2b4481cd84
  121                   DEBUG  I  d20 4026a10d00000000  d21 400e0c0400000000
  121                   DEBUG  I  d22 0000000000000000  d23 401b2981597f8e95
  121                   DEBUG  I  d24 0000000000000000  d25 0000004700000046
  121                   DEBUG  I  d26 0000000000000000  d27 0000000400000004
  121                   DEBUG  I  d28 0000000000000000  d29 0000000000000000
  121                   DEBUG  I  d30 0000000400000004  d31 0000000400000004
  121                   DEBUG  I  scr 80000010
  121                   DEBUG  I  backtrace:
  121                   DEBUG  I  #00  pc 000a0b26  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so (ePub3::ZipFileByteStream::Seek(unsigned int, std::_Ios_Seekdir)+29)
  121                   DEBUG  I  #01  pc 000eb1af  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so (Java_org_readium_sdk_android_util_ResourceInputStream_nativeSkip+38)
  121                   DEBUG  I  #02  pc 0001dbcc  /system/lib/libdvm.so (dvmPlatformInvoke+112)
  121                   DEBUG  I  #03  pc 0004e123  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+398)
  121                   DEBUG  I  #04  pc 00026fe0  /system/lib/libdvm.so
  121                   DEBUG  I  #05  pc 0002dfa0  /system/lib/libdvm.so (dvmMterpStd(Thread*)+76)
  121                   DEBUG  I  #06  pc 0002b638  /system/lib/libdvm.so (dvmInterpret(Thread*, Method const*, JValue*)+184)
  121                   DEBUG  I  #07  pc 00060581  /system/lib/libdvm.so (dvmCallMethodV(Thread*, Method const*, Object*, bool, JValue*, std::__va_list)+336)
  121                   DEBUG  I  #08  pc 000605a5  /system/lib/libdvm.so (dvmCallMethod(Thread*, Method const*, Object*, JValue*, ...)+20)
  121                   DEBUG  I  #09  pc 0005528b  /system/lib/libdvm.so
  121                   DEBUG  I  #10  pc 0000d228  /system/lib/libc.so (__thread_entry+72)
  121                   DEBUG  I  #11  pc 0000d3c0  /system/lib/libc.so (pthread_create+240)
  121                   DEBUG  I  stack:
  121                   DEBUG  I  6f39bab8  69ed6100
  121                   DEBUG  I  6f39babc  6b396870  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so
  121                   DEBUG  I  6f39bac0  69ed6100
  121                   DEBUG  I  6f39bac4  6f39bacc
  121                   DEBUG  I  6f39bac8  69b0fa00
  121                   DEBUG  I  6f39bacc  69ed6100
  121                   DEBUG  I  6f39bad0  00000006
  121                   DEBUG  I  6f39bad4  00000000
  121                   DEBUG  I  6f39bad8  00000006
  121                   DEBUG  I  6f39badc  00000010
  121                   DEBUG  I  6f39bae0  69ed6100
  121                   DEBUG  I  6f39bae4  6a644ef8
  121                   DEBUG  I  6f39bae8  00000000
  121                   DEBUG  I  6f39baec  6e523e1c
  121                   DEBUG  I  6f39baf0  6f39bb0c
  121                   DEBUG  I  6f39baf4  6b227b23  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so (ePub3::ZipFileByteStream::Seek(unsigned int, std::_Ios_Seekdir)+26)
  121                   DEBUG  I  #00  6f39baf8  6b39cd4c  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so
  121                   DEBUG  I  6f39bafc  6b2721b1  /data/app-lib/org.readium.sdk.android.launcher-2/libepub3.so (Java_org_readium_sdk_android_util_ResourceInputStream_nativeSkip+40)
  121                   DEBUG  I  #01  6f39bb00  62afd598  /dev/ashmem/dalvik-LinearAlloc (deleted)
  121                   DEBUG  I  6f39bb04  415cfbd0  /system/lib/libdvm.so (dvmPlatformInvoke+116)
  121                   DEBUG  I  #02  6f39bb08  0018619e
  121                   DEBUG  I  6f39bb0c  415fb111  /system/lib/libdvm.so
  121                   DEBUG  I  6f39bb10  6e523e0c
  121                   DEBUG  I  6f39bb14  00000001
  121                   DEBUG  I  6f39bb18  00000000
  121                   DEBUG  I  6f39bb1c  42188dd0  /dev/ashmem/dalvik-heap (deleted)
  121                   DEBUG  I  6f39bb20  421743d8  /dev/ashmem/dalvik-heap (deleted)
  121                   DEBUG  I  6f39bb24  41600127  /system/lib/libdvm.so (dvmCallJNIMethod(unsigned int const*, JValue*, Method const*, Thread*)+402)```

**Comment by marcuswu** [comment]
This or something similar happened to me with an epub containing corrupt font files. Changing the epub fixed the issue, but it would be good to not crash.

**Comment by rkwright** [comment]
@marcuswu: Would you be able to share the EPUB with the corrupt font files that caused the crash?  We're always looking for such files.  Thanks.

**Comment by danielweck** [comment]
Issue was addressed in #50
